File: MathAndPrimeNumbers/PrimeInDiagonal/primeDiag.py
from math import sqrt
from typing import List

class Solution:

    def isPrime(self,n):
        if n<=1: return False
        for i in range(2,int(sqrt(n)+1)):
            if n%i == 0: return False
        return True

    def diagonalPrime(self, nums: List[List[int]]) -> int:
        maxPrime = 0
        n = len(nums)
        for i in range(n):
            if self.isPrime(nums[i][i]):
                maxPrime = max(maxPrime,nums[i][i])
            
            if self.isPrime(nums[i][n-i-1]):
                maxPrime = max(maxPrime,nums[i][n-i-1])
            
        return maxPrime

File: MathAndPrimeNumbers/PrimeInDiagonal/primeDiag.cpp
#include <vector>
#include <cmath>
using namespace std;

class Solution {
public:
    bool isPrime(int n) {
        if(n<=1)return false;
        // code here
        for(int i=2;i<=sqrt(n);i++){
            if(n%i == 0)return false;
        }
        return true;
    }
    int diagonalPrime(vector<vector<int>>& nums) {
        int maxPrime = 0;
        int n = nums.size();
        for(int i=0;i<n;i++){
            if(isPrime(nums[i][i])){
                maxPrime = max(maxPrime,nums[i][i]);
            }
            if(isPrime(nums[i][n-i-1])){
                maxPrime = max(maxPrime,nums[i][n-i-1]);
            }
        }
        return maxPrime;
    }
};